src/search/SearchScreen.js

```diff
@@ -59,8 +59,7 @@ class SearchScreen extends Component {
     const noResults = !!this.query && !isFetching && !messages.length;
 
     return (
-      <Screen title="Search" keyboardAvoiding>
-        <SearchInput onChange={this.handleQueryChange} />
+      <Screen title="Search" keyboardAvoiding isSearchBarApplied searchBarOnChange={this.handleQueryChange}>
```

Comment:
Similarly, I would create another component called `SearchScreen` here instead of reusing `Screen` for multiple purposes